### ISSUE TITLE
[visualization] Custom FOV for RangeImage::createFromPointCloud()

### ISSUE THREAD

**Issue by kaesesahnetorte** [PointCloudLibrary/pcl#5289]
I would love to be able to set a custom FOV (field of view) for RangeImage::createFromPointCloud().

**Current Behavior**
the Function crops the field of view just so all the points within the pointcloud are included in the image.
In an image stream with lots of invalid measurements (distance -> infinity) the field of view (size and direction) of the image will vary.
The offset from camera center direction is then saved in image_offset_x and image_offset_y.

**desired behavior**
The user can set the field of view explicitly.
The funciton then renders the image using this FOV, generating an image with expected size.
Points from the pointcloud that are not within the FOV are simply discarded.

**Comment by mvieth** [comment]
Are you sure the [current interface for createFromPointCloud](https://pointclouds.org/documentation/classpcl_1_1_range_image.html#ae3a8a2def6dde7446b9fc456f7992aa9) does not offer what you need? Especially with the parameters `angular_resolution_*`, `max_angle_*`, and `sensor_pose`?

**Comment by kaesesahnetorte** [comment]
Yes, those do NOT solve my problem.
**angular_resolution_*** does what is reasonable.
**max_angle_*** limits the FOV, but through the cropping process the resulting image can end up smaller than defined here.
**sensor_pose** just defines the center heading, from which **image_offset_x** and **image_offset_y** are calculated.

**Comment by mvieth** [comment]
Then probably the call to `cropImage` is what you don't want: https://github.com/PointCloudLibrary/pcl/blob/master/common/include/pcl/range_image/impl/range_image.hpp#L138
How did you install PCL? Would it be possible for you to compile PCL from source with `cropImage` and `recalculate3DPointPositions` removed? Alternatively, you can probably do those changes directly in `range_image.hpp` in your existing PCL installation since `createFromPointCloud` is not compiled when building PCL, but when building your project.
When we are sure that this cropping is what you want to disable, we can consider adding another parameter to `createFromPointCloud`

**Comment by mvieth** [comment]
@kaesesahnetorte  We could turn `cropImage` and `recalculate3dPointPositions` off in the code if the `border_size` parameter of `createFromPointCloud` has the value `std::numeric_limits<int>::min()`. What do you think? This way, we don't need another parameter for this already parameter-rich function, and users have the option to turn the cropping off.

**Comment by kaesesahnetorte** [comment]
Sounds great!

**Comment by kaesesahnetorte** [comment]
IMO I would just make a new parameter though. It seems more intuitive to use to me.

**Comment by kaesesahnetorte** [comment]
Or another function

**Comment by mvieth** [comment]
The function already has 10 parameters, so I would rather avoid adding another one. Since the `border_size` parameter already controls the cropping, I think it is natural to extend its use.
Adding another function which basically does the same thing as `createFromPointCloud` except without cropping also doesn't feel like a good option to me because it would result in some code duplication and might cause confusion for users, which function to use.

**Comment by kaesesahnetorte** [comment]
Good point, that border is not needed for the new functionality. Then I guess this makes sense.
